internal/commands/user/flags.go

```diff
@@ -35,5 +51,101 @@ var (
 		providerTypeGoogle,
 		providerTypeAnonymous,
 		providerTypeCustom,
+		providerTypeApple,
+		providerTypeCustomFunction,
+	}
+)
+
+// statusType enumset
+type statusType string
+
+// String returns the status type display
+func (st statusType) String() string { return string(st) }
+
+// Type returns the statusType type
+func (st statusType) Type() string { return "string" }
+
+// Set validates and sets the status type value
+func (st *statusType) Set(val string) error {
+	newStatusType := statusType(val)
+
+	if !isValidStatusType(newStatusType) {
+		return errInvalidStatusType
+	}
+
+	*st = newStatusType
+	return nil
+}
+
+// set of statusType(s)
+const (
+	statusTypeEmpty     statusType = ""
+	statusTypeConfirmed statusType = "confirmed"
+	statusTypePending   statusType = "pending"
+)
+
+// statusType error msg
+var (
+	errInvalidStatusType = func() error {
+		allStatusTypes := []string{statusTypeConfirmed.String(), statusTypePending.String()}
+		return fmt.Errorf("unsupported value, use one of [%s] instead", strings.Join(allStatusTypes, ", "))
+	}()
+)
+
+// isValidStatusType checks if a valid statusType
+func isValidStatusType(st statusType) bool {
+	switch st {
+	case
+		statusTypeEmpty, // allow statusType to be optional
+		statusTypeConfirmed,
+		statusTypePending:
+		return true
+	}
+	return false
+}
+
+// userStateType is a Realm application user state
+type userStateType string
+
+// String returns the user state string
+func (us userStateType) String() string { return string(us) }
+
+// Type returns the user state type
+func (us userStateType) Type() string { return flags.TypeString }
+
+// Set validates and sets the user state value
+func (us *userStateType) Set(val string) error {
+	newUserStateType := userStateType(val)
+
+	if !isValidUserStateType(newUserStateType) {
+		return errInvalidUserStateType
 	}
+
+	*us = newUserStateType
+	return nil
+}
+
+// set of supported user state values
+const (
+	userStateTypeEmpty    userStateType = ""
+	userStateTypeEnabled  userStateType = "enabled"
+	userStateTypeDisabled userStateType = "disabled"
+)
+
+var (
+	errInvalidUserStateType = func() error {
+		allUserStateTypeTypes := []string{userStateTypeEnabled.String(), userStateTypeDisabled.String()}
+		return fmt.Errorf("unsupported value, use one of [%s] instead", strings.Join(allUserStateTypeTypes, ", "))
+	}()
 )
+
+func isValidUserStateType(us userStateType) bool {
+	switch us {
+	case
+		userStateTypeEmpty, // allow state to be optional
+		userStateTypeEnabled,
+		userStateTypeDisabled:
+		return true
+	}
+	return false
+}
```

Comment:
Initially separate from realm.UserState for interactivity default for flag, since not an option for api values. @makesitgo or anyone else say the word and I'll revert back to using realm.UserState

Reply:
I think we can probably merge all of this in with `realm.UserState`, check out `realm.Location`/`realm.DeploymentModel` for similar cases (I think).  We do allow `LocationNil` and such there as well